tests/unit/s2n_pq_kem_test.c

```diff
@@ -54,6 +54,23 @@ int main()
 {
     BEGIN_TEST();
 
+#if defined(OPENSSL_IS_AWSLC) && defined(AWS_LC_API_VERSION)
+    const unsigned long lc_vers = awslc_api_version_num();
+#else
+    const unsigned long lc_vers = SSLeay();
+#endif
+
+    /* AWS-LC did not always expose an API version. Now that it does, it only occupies the
+     * lower byte of an unsigned long. OpenSSL and AWS-LC expose _OpenSSL_'s version
+     * number through SSLeay(), and OpenSSL's version occupies most bytes of the long.*/
+    bool is_lc_vers_api_vers = (lc_vers & 0xffffffffffffff00L) == 0;
```

Comment:
maybe I missed it but were you able to check if this bitwise math is correct?

Reply:
yes, &'ing the mask results in `0` for all values that fit into a byte, and nonzero thereafter.

```
$ python3
Python 3.10.6 (main, Sep 28 2022, 16:04:28) [Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 0xffffffffffffff00 & 21
0
>>> 0xffffffffffffff00 & 255
0
>>> 0xffffffffffffff00 & 256
256
>>> 0xffffffffffffff00 & 0x1010107f
269488128
```

`0x1010107f` is the SSLeay() version defined [here](https://github.com/aws/aws-lc/blob/main/include/openssl/base.h#L197-L199).

Reply:
Thanks for testing that and a good check!

Some nits:
- is there a better name for the variable `is_lc_vers_api_vers`. Maybe `is_awslc_api_version`
- an explicit comment that states what you are checking (added a comment above)

Reply:
Is this worth a define in a header somewhere, like we do for openssl? https://github.com/aws/s2n-tls/blob/main/crypto/s2n_openssl.h#L41-L43 This kind of math can be both hard to read and easy to get wrong :\

Reply:
i think this test would be a lot simpler if we conditional its compilation on the presence of AWS_LC_API_VERSION. this way, we don't need to any bit twiddling and still get the necessary AWS-LC Kyber coverage when built against a sufficiently modern AWS-LC.